dashboard/app/controllers/api/v1/teacher_feedbacks_controller.rb

```diff
@@ -0,0 +1,43 @@
+class Api::V1::TeacherFeedbacksController < Api::V1::JsonApiController
+  authorize_resource
+  load_resource only: :create
+
+  # Use student_id, level_id, and teacher_id to lookup the feedback for a student on a particular level and provide the
+  # most recent feedback left by that teacher
+  def show_feedback_from_teacher
+    @feedback = TeacherFeedback.where(
+      student_id: params[:student_id],
+      level_id: params[:level_id],
+      teacher_id: params[:teacher_id]
+    ).latest
+    render json: @feedback
+  end
+
+  # Use student_id and level_id to lookup the most recent feedback from each teacher who has provided feedback to that
+  # student on that level
+  def show_feedback_for_level
+    @level_feedbacks = TeacherFeedback.where(
+      student_id: params[:student_id],
+      level_id: params[:level_id]
+    ).latest_per_teacher
+
+    render json: {feedbacks: @level_feedbacks}
+  end
```

Comment:
I have some minor nit-picky feedback here on name and format. Bear with me...

 1. `show_feedback_from_teacher` serializes a single `TeacherFeedback` as the entire JSON response.
 2. `show_feedback_for_level` serializes a list inside a `feedbacks` container. 

I'd rather either include the single response from (1) in a `feedback` container, or remove the `feedbacks` container from (2) and return the array at the root so they're consistent. I don't see why one has a container and the other doesn't. Also, (2) should probably be pluralized, `show_feedbacks_for_level`.

Furthermore, typically [show](http://guides.rubyonrails.org/routing.html#crud-verbs-and-actions) is used to display a single item (usually by id). I think a different verb here would be better, even something like `get`. This could even be an `index`, with optional (or even required) filtering params.

This brings up another point. What do we want to do if these params are not provided? Currently I'm pretty sure it will just turn up nothing because for example filtering `where(student_id: nil)` is always empty since that id is required. Same with the others. Maybe that's fine, but it's not entirely intuitive. It might be better to either require the params and return bad request when they're missing, or allow them to be missing and treat it like an index with a filter. In other words, allow the API to work with or without filter params, but the caller will always provide them. 

The `for_level` part of (2) is unnecessary and confusing since the other action also accepts a level_id.

Reply:
Please add a `TeacherFeedbackSerializer` and use that for serialization so we can select which attributes to include. By default, this is calling `to_json` on the model instance and might include extra attributes we don't care about returning.

Reply:
I added the feedbacks container to solve a problem I was having returning an array of records. When I substitute `render json: @level_feedbacks` (and appropriately adjust the controller tests), I get a bunch of errors. Is there a syntax thing I'm missing to return an array of records?